extensions/amp-story/0.1/page-advancement.js

```diff
@@ -286,7 +287,8 @@ class ManualAdvancement extends AdvancementConfig {
         ((1 - NEXT_SCREEN_AREA_RATIO) * offsetWidth);
     const nextScreenAreaMax = offsetLeft + offsetWidth;
 
-    if (event.pageX >= nextScreenAreaMin && event.pageX < nextScreenAreaMax) {
+    if ((event.pageX >= nextScreenAreaMin && event.pageX < nextScreenAreaMax)
```

Comment:
Adding knowledge to `ManualAdvancement` about desktop mode tangles things, IMO.

How about reversing the logic flow? `ManualAdvancement` could pass along that it comes from an `areaClick` or something like that, and then `AmpStory` can reject it.

```javascript
// ManualAdvancement
if (event.pageX >= nextScreenAreaMin && event.pageX < nextScreenAreaMax) {
   this.onAdvance(/* opt_isFromAreaClick */ true);
}
```

```javascript
// AmpStory

buildCallback() {
  //...
  this.advancement_.addAdvanceListener(isFromAreaClick => {
    // TODO(alanorozco): opt_isAutomatic is incorrect in this case, should be passed from
    // advancement class. Ideally refactor into AdvancementType enum (`AUTO/AREA_CLICK/BUTTON_CLICK`).
    this.next(/* opt_isAutomaticAdvance */ !isFromAreaClick, isFromAreaClick));
  });
}

// next() passes arg thru to switchTo_()

// ...
switchTo_(targetPageId, opt_isFromAreaClick) {
   const payload = {targetPageId, isFromAreaClick: !!opt_isFromAreaClick};

  // ...
}
```

Reply:
@alanorozco I completely agree with this.

However in the above approach i'll still be able to either let it go forward or ignore it. 
There would still be a probability that I end up calling `previous` from `ManualAdvancement` and in the end call `.next` from inside it. which would again be ugly.

Can we instead remove `onAdvance` / `onPrevious` from AdvancementConfiguration and just have `onNavigate`? This ways `AdvancementConfiguration` and its subclasses can pass the type of navigation and its metadata. 

e.g. `onNavigate({NAVIGATION.MANUAL/TIMER/MEDIA, TAP_AREA: 0.8})` and then `AMP-STORY-PAGE` can decide whether to call `.next` or `.previous`

Reply:
As discussed offline: Best idea is to add a new `NAVIGATION` event that specifies direction and both target IDs.